folium/features.py

```diff
@@ -602,25 +604,124 @@ def get_bounds(self):
+        properties like font and colors in a div element in the tooltip.
+    sticky: bool, default True
+        Whether the tooltip should follow the mouse.
+    **kwargs: Assorted.
+        These values will map directly to the Leaflet Options. More info
+        available here: https://leafletjs.com/reference-1.3.0.html#tooltip
+
+    Examples
+    --------
+    # Provide fields and aliases, with Style.
+    >>> Tooltip(
+    >>>     fields=['CNTY_NM','census-pop-2015','census-md-income-2015'],
+    >>>     aliases=['County','2015 Census Population', '2015 Median Income'],
+    >>>     labels=True,
+    >>>     localize=True,
+    >>>     style=('background-color: grey; color: white; font-family:'
+    >>>            'courier new; font-size: 24px; padding: 10px;')
+    >>> )
+    # Provide fields, with labels off, and sticky True.
+    >>> Tooltip(fields=('CNTY_NM',), labels=False)
     """
-    if isinstance(tooltip, Tooltip):
-        if tooltip.fields:
-            for value in tooltip.fields:
-                assert value in keys, ("The value {} is not available in {}"
-                                       .format(value, keys))
-        return tooltip
-    else:
-        return create_tooltip(tooltip)
+    _template = Template(u"""
+        {% macro script(this, kwargs) %}
+        {{ this._parent.get_name() }}.bindTooltip(
+            function(layer){
+            // Convert non-primitive to String.
+            let handleObject = (feature)=>typeof(feature)=='object' ? JSON.stringify(feature) : feature;
+            let fields = {{ this.fields }};
+            {% if this.aliases %}
+            let aliases = {{ this.aliases }};
+            {% endif %}
+            return '<table{% if this.style %} style="{{this.style}}"{% endif%}>' +
+            String(
+                fields.map(
+                columnname=>
+                    `<tr style="text-align: left;">{% if this.labels %}
+                    <th style="padding: 4px; padding-right: 10px;">{% if this.aliases %}
+                        ${aliases[fields.indexOf(columnname)]
+                        {% if this.localize %}.toLocaleString(){% endif %}}
+                    {% else %}
+                    ${ columnname{% if this.localize %}.toLocaleString(){% endif %}}
+                    {% endif %}</th>
+                    {% endif %}
+                    <td style="padding: 4px;">${handleObject(layer.feature.properties[columnname])
+                    {% if this.localize %}.toLocaleString(){% endif %}}</td></tr>`
+                ).join(''))
+                +'</table>'
+            }, {{ this.options }});
+        {% endmacro %}
+        """)
+
+    def __init__(self, fields, aliases=None, labels=True,
+                 localize=False, style=None, sticky=True, **kwargs):
+        super(GeoJsonTooltip, self).__init__(
+            text='', style=style, sticky=sticky, **kwargs
+        )
+        self._name = "Tooltip"
```

Comment:
Not really important, but this line is redundant since it is already defined in `Tooltip`.